packages/aws-cdk-lib/aws-ec2/lib/launch-template.ts

```diff
@@ -786,6 +817,13 @@ export class LaunchTemplate extends Resource implements ILaunchTemplate, iam.IGr
     }
   }
 
+  /**
+   * @internal
+   */
+  public _addIamInstanceProfileArn(iamInstanceProfileArn: string) {
```

Comment:
I feel like we should just bit the bullet and create a `InstanceProfile` L2. It would be a super simple L2 and would allow us to just pass it in through properties of the launch template

Reply:
I created an `InstanceProfile` L2 and have the functionality we want working using the L2. I need to finish testing the `InstanceProfile` L2 and then I will push these changes.